imagescan/subscriber_test.go

```diff
@@ -182,46 +199,64 @@ func TestSubscriber(t *testing.T) {
 		scanner := &mockImageScanner{}
 		delta := NewDeltaState(nil)
 		sub := NewSubscriber(log, cfg, scanner, 21, delta)
-		ctx, cancel := context.WithTimeout(ctx, 50*time.Millisecond)
+		ctx, cancel := context.WithCancel(ctx)
 		defer cancel()
 
-		sub.OnAdd(node1)
-		sub.OnAdd(node2)
-		sub.OnAdd(argoReplicaSet)
-		sub.OnAdd(argoPod1)
-		sub.OnAdd(argoPod2)
-		sub.OnAdd(nginxPod1)
-		sub.OnAdd(nginxPod2)
-		err := sub.Run(ctx)
-		r.True(errors.Is(err, context.DeadlineExceeded))
-
-		sort.Slice(scanner.imgs, func(i, j int) bool {
-			return scanner.imgs[i].ImageName < scanner.imgs[j].ImageName
+		// Simulate concurrent deltas update.
+		go func() {
```

Comment:
this is single goroutine, for concurrency you should run multiple goroutines

Reply:
wouldn't that just break everything since there's a single subscriber without locks now

Reply:
Check subscriber.Run, it is inside separate go routine so we have two goroutines here. Or you are not talking about this unit test?